apps/web/src/utils/storage/companyDataStorage.ts

```diff
@@ -27,6 +27,13 @@ export default class CompanyDataStorage {
     return this._data.value[CompanyDataStorageKey.Company] !== null
   }
 
+  public static hasSiret() {
+    return (
+      this.hasCompanyData() &&
+      Object.hasOwn(this._data.value[CompanyDataStorageKey.Company] as EstablishmentFront | ManualCompanyData, 'siret') &&
+      (this._data.value[CompanyDataStorageKey.Company] as EstablishmentFront).siret !== null
```

Comment:
Praise : comme je le disais, sympa comme feature

Question : le cast est il obligatoire ? je me demande si ça ne peut pas créer des erreur alors même que cette vérification est hors périmètre de la méthode de mon point de vue.

Question : Dans le back, j'imaginai pouvoir unifier les types `EstablishmentFront | ManualCompanyData` suite aux dernières modifs sur la modale. Vous pensez que ce sera faisable/souhaitable à terme ?

Reply:
carrément pour unifier les types, une fois qu'on aura la recherche naf et la ville on aura quasiment pas de différence entre les deux

Reply:
this._data.value[CompanyDataStorageKey.Company] n'est pas censé être null mais j'ai une erreur dans mon IDE car hasOwn n'accepte pas `null` en entrée, je peux peut-être plutôt proposer qqchose comme ça en attendant :
 ```typescript
   public static hasSiret() {
    if (!this._data.value[CompanyDataStorageKey.Company]) return false

    return (
      Object.hasOwn(this._data.value[CompanyDataStorageKey.Company], 'siret') &&
      (this._data.value[CompanyDataStorageKey.Company] as EstablishmentFront).siret !== null
    )
  }
  ```